### ISSUE TITLE
Support `@JsonDiscriminator`

### ISSUE THREAD

**Issue by aSemy** [adamko-dev/kotlinx-serialization-typescript-generator#4]
kxs-ts-gen should detect `@JsonDiscriminator`, and use the `@SerialName` as a hardcoded value

```kotlin
@Serializable
@JsonDiscriminator("type")
sealed class SomeClass {
  val id: String
}

@SerialName("named")
@Serializable
data class NamedSomeClass(
  override val id: String,
  val name: String,
) : SomeClass()
```

Should convert to something like...

```typescript
interface SomeClass<T extends "named"> {
  id: string;
  type: T
}

interface NamedSomeClass extends SomeClass {
  id: string;
  type: "named"
}
```
